source/engine/ui/editor/impl/ComponentWidgets.cpp

```diff
@@ -133,6 +133,79 @@ constexpr auto useGravityProp  = nc::ui::Property{ &T::UseGravity,     &T::SetUs
 constexpr auto isKinematicProp = nc::ui::Property{ &T::IsKinematic,    &T::SetIsKinematic, "isKinematic" };
 } // namespace physics_body_ext
 
+namespace particle_emitter_ext
+{
+using T = nc::graphics::ParticleEmitter;
+
+#define DECLARE_SETTER(name, memberAccess)     \
+constexpr auto name = [](auto& obj, auto& v)   \
+{                                              \
+    auto info = obj.GetInfo();                 \
+    info.memberAccess = v;                     \
+    obj.SetInfo(std::move(info));              \
+};
+
+constexpr auto getMaxParticleCount = [](auto& obj) { return obj.GetInfo().emission.maxParticleCount; };
+constexpr auto getInitialEmissionCount = [](auto& obj) { return obj.GetInfo().emission.initialEmissionCount; };
+constexpr auto getperiodicEmissionCount = [](auto& obj) { return obj.GetInfo().emission.periodicEmissionCount; };
+constexpr auto getPeriodicEmissionFrequency = [](auto& obj) { return obj.GetInfo().emission.periodicEmissionFrequency; };
+constexpr auto getLifetime = [](auto& obj) { return obj.GetInfo().init.lifetime; };
```

Comment:
Some of these are alphabetized, others are not.
Also, "GetperiodicEmissionCount" needs to be capitalized